napari/plugins/hook_specifications.py

```diff
@@ -496,3 +496,33 @@ def napari_experimental_provide_dock_widget() -> Union[
     >>> def napari_experimental_provide_dock_widget():
     ...     return threshold
     """
+
+
+@napari_hook_specification(firstresult=True)
+def napari_experimental_provide_theme() -> Tuple[
+    List[str], List[str], Dict[str, Dict[str, str]]
+]:
+    """Provide GUI theme metadata that can be used when generating theme .py files.
+
+    This hook specification is marked as experimental as the API or how the returned
+    value is handled may change here more frequently than the rest of the codebase.
+
+    Returns
```

Comment:
I might want to reorder this tuple here so that `color_dict` comes first, then `qss_files`, and then `svg_paths` as I see `color_dict` as the simplest most widely accessible option. It's not to much to ask folks who don't want to provide the other things to put in empty lists, but we could also think about just allowing the `Dict[str, Dict[str, str]]` for people who only want to return a `color_dict`. Not sure I need that, but putting it out there

Reply:
Actually thinking further I wonder if we want to have two hookspecs here. One for providing the color dictionary and one for providing the custom qss/ icon svgs.

I have two reasons for this. One is that that the color dict is independent of Qt so could be very stable (and be reused if we ever did another front end). And two at any moment in time you only want to have one theme/ color_dict be active, and these choices could populate the settings and preferences (cc @goanpeca @ppwadhwa according to Lia's design). These are then thought of in an "exclusive" manner, at any one time you only use the output of one plugin or our builtin.

In contrast, the additional qss sheets and icons are something that should be thought as additive, i.e. if you have 5 plugins the offering from all 5 should now be available, though one might need to think about conflicts here ...

Not sure if this is really the right approach either, but curious what others think too.

Reply:
As I mentioned above, I think I will split it into two separate hooks since it's probably an overkill to always have to provide the icons/stylesheets if only color scheme is being changed. 

> In contrast, the additional qss sheets and icons are something that should be thought as additive, i.e. if you have 5 plugins the offering from all 5 should now be available, though one might need to think about conflicts here ...

Indeed, this is my concern that if multiple plugins are providing icons/stylesheets/themes then it could become a bit of a mess. Perhaps they could be ordered like the `reader/writer/etc` plugins but that might not be that elegant.

Reply:
Yeah, one concrete use case for adding icons is to let plugins add their own icon/ and "about" text description that we generate a widget from. This might be a new hookspec as we could generate the widget for them and show it at various places in the app, but wanted to mention it. I know @tlambert03 and I have discussed it in other places (cc @neuromusic)